lib/src/publication/local.dart

```diff
@@ -27,4 +33,39 @@ class LocalTrackPublication<T extends LocalTrack> extends TrackPublication<T> {
 
   /// Unmute the track associated with this publication
   Future<void> unmute() async => await track?.unmute();
+
+  @internal
+  void updatePublishingLayers(List<lk_rtc.SubscribedQuality> layers) async {
+    //
+    final params = track?.sender?.parameters;
+    if (params == null) return;
+
+    final encodings = params.encodings;
+    if (encodings == null) return;
+
+    bool didChange = false;
+
+    for (final encoding in encodings) {
+      final layer = layers.firstWhereOrNull((e) {
+        // If there is exact match, use it
+        if (e.quality.toRid() == encoding.rid) return true;
+        // Use first layer found if not simulcast
+        if (encoding.rid == null && encodings.length == 1) return true;
+        return false;
+      });
+      if (layer != null && encoding.active != layer.enabled) {
+        encoding.active = layer.enabled;
```

Comment:
since Flutter includes web, it'll likely suffer from the same issue as web SDK with FireFox. I think it'd be good to put a comment in here, but it's fine if we don't handle it right now.

Reply:
Is flutter-webrtc using Firefox's WebRTC library for web?

Reply:
they simply translate into equivalent calls from dart to JS. So any browser specific implementation quirks will still remain